packages/language-server/test/completion-items-spec.ts

```diff
@@ -25,28 +29,245 @@ describe("the UI5 language assistant Code Completion Services", () => {
   });
 
   // Return the first part of a tag name suggestion insert text
-  function getTagName(insertText: string | undefined): string | undefined {
-    if (insertText === undefined) {
+  function getTagName(textEdit: TextEdit | undefined): string | undefined {
+    if (textEdit === undefined) {
       return undefined;
     }
-    const result = /^([^> ]*)/.exec(insertText);
+    const result = /^([^> ]*)/.exec(textEdit.newText);
     return result?.[1];
   }
 
+  function assertRangeContains(
+    range: Range,
+    position: Position,
+    description: string
+  ): void {
+    // The range must be in the same line as the position
+    expect(range.start.line, `${description}: range start line`).to.equal(
+      position.line
+    );
+    expect(range.end.line, `${description}: range end line`).to.equal(
+      position.line
+    );
+    expect(
+      range.start.character,
+      `${description}: range start character`
+    ).to.be.at.most(position.character);
+    expect(
+      range.end.character,
+      `${description}: range end character`
+    ).to.be.at.least(position.character);
+  }
+
+  function assertFilterMatches(
+    filterText: string,
+    text: string,
+    description: string
+  ): void {
+    // This is a simple matcher - all characters in text are found in filterText in the same order.
```

Comment:
What do you think about implementing this by dynamically creating a regExp?
It seems more declarative to me.
- /^B.*U.*T.*$/.test("BUTTON) // --> true

```
> "BUT".split("").join(".*")
'B.*U.*T'
```

Reply:
It would be a problem if there are characters that should be escaped (like dots), which will make it more complex than the iterative approach I think. I tried to do it with an array matcher (`to.include.ordered.members`) but that ignores duplicate entries.